test/read-only-handler.spec.ts

```diff
@@ -271,4 +271,41 @@ describe('ReadOnlyHandler', () => {
             doNothing(readOnly.foo);
         }).not.toThrow();
     });
+
+    describe.skip('issue#20 - getOwnPropertyDescriptor', () => {
+        it('should return reactive proxy when property accessed via raw getter', () => {
+            const target = new ReactiveMembrane();
+            const todos = {};
+            Object.defineProperty(todos, 'entry', {
+                get() {
+                    return { foo: 'bar' };
+                },
+                configurable: true
+            });
+
+            const proxy = target.getReadOnlyProxy(todos);
+            const desc = Object.getOwnPropertyDescriptor(proxy, 'entry');
+            const { get } = desc;
+            expect(() => { 
+                get().foo = '';
+            }).toThrow();
+            expect(todos['entry'].foo).toEqual('bar');
+        });
+        it('should return reactive proxy when property accessed via descriptor', () => {
+            const target = new ReactiveMembrane();
+            const todos = {};
+            Object.defineProperty(todos, 'entry', {
+                value : { foo: 'bar' },
+                configurable: true
+            });
+
+            const proxy = target.getReadOnlyProxy(todos);
+            const desc = Object.getOwnPropertyDescriptor(proxy, 'entry');
+            const { value } = desc;
+            expect( () => {
+                value.foo = '';
+            }).toThrow();
+            expect(todos['entry'].foo).toEqual('bar');
+        });
+    })
```

Comment:
aside from assertion to throw, what is the difference between the tests in this file and reactive-handler.spec.ts? Wondering if we are duplicating here

Reply:
They are testing two different APIs. Test cases in reactive-handler.spec.ts is testing that the inner object is wrapped in a reactive proxy. The test cases in this file is testing that the readonly proxy cannot be bypassed.

The testcase name is wrong. Will fix it.

Reply:
Both the tests exploit the same loop hole in the implementation.